src/App.js

```diff
@@ -14,11 +14,135 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: goodsFromServer,
```

Comment:
don't save two arrays

Reply:
Actually I don't understand how the length filter should work then. We always need the original array, otherwise, once we remove short words, we will not be able to return them in any way except reset.

Reply:
You should always work with available list `goodsFromServer` - prepare it in `App` and pass it to `GoodsList`.
Just remember not to mutate original array.

Hint - you can save sort and filter conditions in state.